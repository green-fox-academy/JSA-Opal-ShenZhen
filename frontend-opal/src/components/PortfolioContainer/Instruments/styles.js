import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  instrumentsArea: {
    width: '100%'
  },
  title: {
    fontSize: 16
  },
  header: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    padding: 10
  },
  profileImgContainer: {
    width: 40,
    height: 40
  },
  profileImg: {
    width: '100%',
    height: '100%'
  },
  profileText: {
    marginLeft: 10
  },
  stockExchange: {
    color: '#B2B2B2'
  },
  instruments: {
    backgroundColor: '#DFF1FD',
    paddingLeft: 10,
    paddingRight: 10,
    paddingTop: 10
  },
  instrumentDetail: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginBottom: 10
  },
  instrumentDetailText: {
    fontSize: 16
  },
  instrumentDetailTextGreen: {
    color: '#5FC748'
  },
  buttonContainer: {
    display: 'flex',
    flexDirection: 'row',
    padding: 10
  },
  button: {
    marginLeft: 10,
    marginRight: 10,
    backgroundColor: 'white',
    paddingLeft: 10,
    paddingRight: 10
  }
});
