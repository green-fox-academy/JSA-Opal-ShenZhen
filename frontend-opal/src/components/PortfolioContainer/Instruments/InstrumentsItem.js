import React from 'react';
import { View, Text, Image } from 'react-native';
import PropTypes from 'prop-types';
import { Card, Button } from 'native-base';
import styles from './styles';

function InstrumentsItem({ instrumentItem }) {
  const { company, stockExchange, profileImg, amount, marketValue } = instrumentItem;

  return (
    <View>
      <Card>
        <View style={styles.header}>
          <View style={styles.profileImgContainer}>
            <Image style={styles.profileImg} source={profileImg} />
          </View>
          <View style={styles.profileText}>
            <Text>{company}</Text>
            <Text style={styles.stockExchange}>{stockExchange}</Text>
          </View>
        </View>

        <View style={styles.instruments}>
          <View style={styles.instrumentDetail}>
            <Text style={styles.instrumentDetailText}>Positions:</Text>
            <Text style={styles.instrumentDetailText}>{amount}</Text>
          </View>
          <View style={styles.instrumentDetail}>
            <Text style={styles.instrumentDetailText}>Market Value:</Text>
            <Text style={styles.instrumentDetailText}>{marketValue}</Text>
          </View>
          <View style={styles.instrumentDetail}>
            <Text style={styles.instrumentDetailText}>Unrlzd P/L %:</Text>
            {/* <Text style={[styles.instrumentDetailText, styles.instrumentDetailTextGreen]}>
              {unrlzedPLPercentage}
            </Text> */}
          </View>
          <View style={styles.instrumentDetail}>
            <Text style={styles.instrumentDetailText}>Unrlzd P/L:</Text>
            {/* <Text style={[styles.instrumentDetailText, styles.instrumentDetailTextGreen]}>
              {unrlzedPL}
            </Text> */}
          </View>
        </View>

        <View style={styles.buttonContainer}>
          <Button style={styles.button}>
            <Text>TRADE</Text>
          </Button>
          <Button style={styles.button}>
            <Text>DETAILS</Text>
          </Button>
        </View>
      </Card>
    </View>
  );
}

InstrumentsItem.propTypes = {
  instrumentItem: PropTypes.shape({
    company: PropTypes.string,
    stockExchange: PropTypes.string,
    profileImg: PropTypes.any,
    amount: PropTypes.number,
    marketValue: PropTypes.number
    // unrlzedPLPercentage: PropTypes.string,
    // unrlzedPL: PropTypes.number
  }).isRequired
};

export default InstrumentsItem;
