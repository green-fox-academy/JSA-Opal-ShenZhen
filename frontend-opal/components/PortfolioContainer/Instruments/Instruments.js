import React from 'react';
import { SafeAreaView, Text, FlatList } from 'react-native';
import PropTypes from 'prop-types';
import InstrumentsItem from './InstrumentsItem';
import styles from './styles';

const Instruments = ({ InstrumentList }) => (
  <SafeAreaView style={styles.instrumentsArea}>
    <Text style={styles.title}>Instruments</Text>
    <FlatList
      data={InstrumentList}
      renderItem={({ item }) => (
        <InstrumentsItem InstrumentItem={item} />
      )}
      keyExtractor={(item) => item.stockExchange + item.company}
    />
  </SafeAreaView>
);

Instruments.propTypes = {
  InstrumentList: PropTypes.arrayOf(
    PropTypes.shape({
      company: PropTypes.string,
      stockExchange: PropTypes.string,
    }),
  ).isRequired,
};

export default Instruments;
