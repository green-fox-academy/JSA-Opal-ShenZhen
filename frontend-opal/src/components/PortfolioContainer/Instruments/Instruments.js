import React from 'react';
import { SafeAreaView, Text, FlatList } from 'react-native';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Card, CardItem, Spinner, DeckSwiper, Container } from 'native-base';
import InstrumentsItem from './InstrumentsItem';
import styles from './styles';

const Instruments = ({ portfolio }) => {
  return (
    <CardItem>
      <SafeAreaView style={styles.instrumentsArea}>
        <Card transparent>
          <Text style={styles.title}>Instruments</Text>
        </Card>
        {!portfolio[0] ? (
          <Spinner color="#e17055" />
        ) : (
          <FlatList
            data={portfolio}
            renderItem={({ item }) => <InstrumentsItem instrumentItem={item} />}
            keyExtractor={item => `${item.id}`}
          />
        )}
      </SafeAreaView>
    </CardItem>
  );
};

const mapStateToProps = state => ({
  portfolio: state.portfolio
});

export default connect(mapStateToProps, null)(Instruments);

Instruments.defaultProps = {
  portfolio: []
};

Instruments.propTypes = {
  portfolio: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      user_id: PropTypes.number,
      symbol: PropTypes.string,
      amount: PropTypes.number,
      company: PropTypes.string,
      stockExchange: PropTypes.string,
      marketValue: PropTypes.number,
      profileImg: PropTypes.any,
      sector: PropTypes.string,
      news: PropTypes.any
    })
  )
};
