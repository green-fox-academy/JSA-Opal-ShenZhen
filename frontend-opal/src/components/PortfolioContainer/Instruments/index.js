import Instruments from './Instruments';

export default Instruments;
