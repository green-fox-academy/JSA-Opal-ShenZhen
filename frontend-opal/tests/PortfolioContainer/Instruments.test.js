import React from 'react';
import renderer from 'react-test-renderer';
import Instruments from '../../src/components/PortfolioContainer/Instruments';

const instrumentsList = [{
  company: 'FB - Facebook Inc.',
  stockExchange: 'NASDAQ',
  /* eslint-disable-next-line */
  profileImg: require('../../src/components/PortfolioContainer/Instruments/images/avatarPlaceholder.png'),
  positions: 136,
  marketValue: 11863.28,
  unrlzedPLPercentage: '4.5%',
  unrlzedPL: 510.86,
}];

describe('<Instruments />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Instruments InstrumentList={instrumentsList} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
