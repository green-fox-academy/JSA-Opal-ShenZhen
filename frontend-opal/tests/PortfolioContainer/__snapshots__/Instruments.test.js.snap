// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Instruments /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderRadius": 2,
      "flexDirection": "row",
      "paddingBottom": 10,
      "paddingLeft": 15,
      "paddingRight": 15,
      "paddingTop": 10,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "transparent",
          "borderBottomWidth": 0,
          "borderColor": "#ccc",
          "borderLeftWidth": 0,
          "borderRadius": 2,
          "borderRightWidth": 0,
          "borderTopWidth": 0,
          "elevation": null,
          "flexWrap": "nowrap",
          "marginBottom": 5,
          "marginLeft": 2,
          "marginRight": 2,
          "marginTop": 5,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
        }
      }
      transparent={true}
    >
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
      >
        Instruments
      </Text>
    </View>
    <ActivityIndicator
      animating={true}
      color="#e17055"
      hidesWhenStopped={true}
      size="large"
      style={
        Object {
          "height": 80,
        }
      }
    />
  </RCTSafeAreaView>
</View>
`;
